src/Code/Enemies/Hitbox.gd

```diff
@@ -1,5 +1,37 @@
 extends Area2D
 
+var invincible = false setget set_invincible
+
+onready var timer = $Timer
+onready var collisionShape = $Shape
+
+signal invincibility_started
+signal invincibility_ended
 
 func set_target(t: Node2D):
 	get_parent().set_target(t)
+
+func set_invincible(value):
+	print("Setting Invincible!")
+	invincible = value
+	if invincible == true:
+		emit_signal("invincibility_started")
+	else:
+		emit_signal("invincibility_ended")
+
+func start_invincibility(duration):
+	self.invincible = true
+	timer.start(duration)
+
+func _on_Timer_timeout():
```

Comment:
I don't understand why timer and hitbox are cap

Reply:
These are Signal Functions, and if you see this format, assume so in the future as these aren't worth the time.